src/main/java/edu/kit/hci/soli/domain/LoginStateModel.java

```diff
@@ -1,21 +1,9 @@
 package edu.kit.hci.soli.domain;
 
+import org.springframework.lang.Nullable;
 import org.springframework.security.web.csrf.CsrfToken;
 
-public class LoginStateModel {
-    public String name;
-    public long visits;
-    public Kind kind;
-    public CsrfToken csrfToken;
-
-
-    public LoginStateModel(String name, long visits, Kind kind, CsrfToken csrfToken) {
-        this.name = name;
-        this.visits = visits;
-        this.kind = kind;
-        this.csrfToken = csrfToken;
-    }
-
+public record LoginStateModel(String name, Kind kind, CsrfToken csrfToken, @Nullable User user) {
```

Comment:
Making this a record is quite a bit cleaner, especially since mutability is explicitly undesirable for this layer.

Reply:
this should be move to the dto package imo, since it is not related to the db model